CRM/Report/Form.php

```diff
@@ -1140,6 +1140,27 @@ public function cleanUpTemporaryTables() {
     }
   }
 
+  /**
+   * Create a temporary table.
+   *
+   * This function creates a table AND adds the details to the developer tab & $this->>temporary tables.
+   *
+   * @todo improve presentation on the developer tab since CREATE TEMPORARY is removed.
+   *
+   * @param string $identifier
+   * @param $sql
+   * @param bool $isTrueTemporary
+   *   Is this a mysql temporary table or temporary in a less technical sense.
+   *
+   * @return string
+   */
+  public function createTemporaryTable($identifier, $sql, $isTrueTemporary = TRUE) {
```

Comment:
@totten does this commit seem correct to you - the goal of the PR is just to get the report under testing. Tear down of the temptable is a blocker after the other issues are resolved - I feel like a helper on the class function makes sense as a) the sql goes onto the developer tab & b) it works with the existing cleanup mechanism - however, you might have some other thoughts.

Note this is an interim cleanup

Reply:
(also the above should be a non-blocking question - we can iterate & improve)